src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -1168,108 +1168,181 @@ def test_ps_s3_notification_errors_on_master():
+        s3_notification_conf = PSNotificationS3(conn, bucket_name, topic_conf_list)
+        response, status = s3_notification_conf.set_config()
+        assert_equal(status/100, 2)
+    elif endpoint_type == 'kafka':
+        # create s3 topic
+        endpoint_address = 'kafka://' + kafka_server
+        # without acks from broker
+        endpoint_args = 'push-endpoint='+endpoint_address+'&kafka-ack-level=broker'
+        topic_conf = PSTopicS3(conn, topic_name, zonegroup, endpoint_args=endpoint_args)
+        topic_arn1 = topic_conf.set_config()
+        endpoint_args = 'push-endpoint='+endpoint_address+'&kafka-ack-level=none'
+        topic_conf2 = PSTopicS3(conn, topic_name2, zonegroup, endpoint_args=endpoint_args)
+        topic_arn2 = topic_conf2.set_config()
+        # create s3 notification
+        notification_name = bucket_name + NOTIFICATION_SUFFIX
+        topic_conf_list = [{'Id': notification_name + '_1', 'TopicArn': topic_arn1,
+                            'Events': []
+                            },
+                           {'Id': notification_name + '_2', 'TopicArn': topic_arn2,
+                            'Events': []
+                            }]
+
+        s3_notification_conf = PSNotificationS3(conn, bucket_name, topic_conf_list)
+        response, status = s3_notification_conf.set_config()
+        assert_equal(status/100, 2)
+    else:
+        return SkipTest('Unknown endpoint type: ' + endpoint_type)
+
+    # create objects in the bucket
+    number_of_objects = 10
     client_threads = []
+    etags = []
+    objects_size = {}
     start_time = time.time()
     for i in range(number_of_objects):
-        key = bucket.new_key(str(i))
         content = str(os.urandom(1024*1024))
-        thr = threading.Thread(target = set_contents_from_string, args=(key, content,))
+        etag = hashlib.md5(content.encode()).hexdigest()
+        etags.append(etag)
+        object_size = len(content)
+        key = bucket.new_key(str(i))
+        objects_size[key.name] = object_size
+        thr = threading.Thread(target=set_contents_from_string, args=(key, content,))
         thr.start()
         client_threads.append(thr)
     [thr.join() for thr in client_threads]
 
     time_diff = time.time() - start_time
-    print('average time for creation + qmqp notification is: ' + str(time_diff*1000/number_of_objects) + ' milliseconds')
+    print('average time for creation + http notification is: ' + str(time_diff*1000/number_of_objects) + ' milliseconds')
 
     print('wait for 5sec for the messages...')
     time.sleep(5)
 
-    # check amqp receiver
+    # check receiver
     keys = list(bucket.list())
-    print('total number of objects: ' + str(len(keys)))
-    receiver1.verify_s3_events(keys, exact_match=True)
-    receiver2.verify_s3_events(keys, exact_match=True)
+    receiver.verify_s3_events(keys, exact_match=True, deletions=False, expected_sizes=objects_size, etags=etags)
+    if endpoint_type != 'http':
```

Comment:
why special handling for http?
* we can change the test to use one receiver in all cases
* or we can have a test with 2 receivers, and use 2 http servers for that listening on 2 different ports

Reply:
http server doesn't have ack levels while **kafka** and **ampq** do have different ack levels,
do you want to delete the second receivers for **kafka** and **amp**q, or add another receiver for http?

Reply:
yes. lets use one topic and receiver in this test.
we can add different ack level test later

Reply:
Sure!